src/amo/components/ShowMoreCard/index.js

```diff
@@ -23,20 +23,46 @@ type UIStateType = {|
   readMoreExpanded: boolean,
 |};
 
+export const truncateToMaxHeight = ({
+  contents,
+  maxHeight,
+  setUIState,
+  uiState,
+}: {|
+  contents: HTMLElement | null,
+  maxHeight: number,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|}) => {
+  if (contents) {
+    // If the contents are short enough they don't need a "show more" link; the
+    // contents are expanded by default.
+    if (uiState.showAllContent && contents.clientHeight >= maxHeight) {
+      setUIState({
+        ...uiState,
+        showAllContent: false,
+      });
+    }
+  }
+};
+
 type DefaultProps = {|
+  _truncateToMaxHeight: typeof truncateToMaxHeight,
   maxHeight?: number,
 |};
 
 type Props = {|
-  ...DefaultProps,
   contentId: number | string | null,
   children: React.Node,
   className?: string,
   header?: React.Node | string,
   id: string,
+  maxHeight?: number,
 |};
 
 type InternalProps = {|
+  _setUIState?: ($Shape<UIStateType>) => void,
```

Comment:
We also need to be able to mock `setUIState` from a test, which is why this has become a new prop.

Reply:
so `_setUIState` only exists to be able to override `setUIState` in tests?  it's pretty weird to me that's a necessity in our testing setup but okay.